2019/test/SuttersTest.java

```diff
@@ -0,0 +1,203 @@
+            public void handleEvent(RobotEvent e) {
+                MineralDetectionEvent event = (MineralDetectionEvent)e;
+                List<Recognition> updatedMinerals = event.minerals;
+                if (updatedMinerals != null) {
+                    numberOfMinerals.setValue(updatedMinerals.size());
+                    if (updatedMinerals.size() == 3) {
+                        int goldMineralX = -1;
+                        int silverMineral1X = -1;
+                        int silverMineral2X = -1;
+                        for (Recognition recognition : updatedMinerals) {
+                            if (recognition.getLabel().equals(LABEL_GOLD_MINERAL)) {
+                                goldMineralX = (int) recognition.getLeft();
+                            } else if (silverMineral1X == -1) {
+                                silverMineral1X = (int) recognition.getLeft();
+                            } else {
+                                silverMineral2X = (int) recognition.getLeft();
+                            }
+                        }
+                        if (goldMineralX != -1 && silverMineral1X != -1 && silverMineral2X != -1) {
+                            if (goldMineralX < silverMineral1X && goldMineralX < silverMineral2X) {
+                                goldMineralPosition.setValue("LEFT");
+                                knockPath = knockLeftPath;
+                            } else if (goldMineralX > silverMineral1X && goldMineralX > silverMineral2X) {
+                                goldMineralPosition.setValue("RIGHT");
+                                knockPath = knockRightPath;
+                            } else {
+                                goldMineralPosition.setValue("CENTER");
+                                knockPath = knockCenterPath;
+                            }
+                        }
+                    }
+                }
+
+            }
+        };
+        mdTask.init(telemetry, hardwareMap);
+        mdTask.setDetectionKind(MineralDetectionTask.DetectionKind.EVERYTHING);
+        this.addTask(mdTask);
+    }
+    */
+
+    private void initializeVuforia (){
+        VuforiaLocalizer.Parameters parameters = new VuforiaLocalizer.Parameters();
+        parameters.vuforiaLicenseKey = VUFORIA_KEY;
+        parameters.cameraName = hardwareMap.get(WebcamName.class, "mineralCamera");
+        vuforia = ClassFactory.getInstance().createVuforia(parameters);
+    }
+
+    private void initializeTfod() {
+        int tfodMonitorViewId = hardwareMap.appContext.getResources().getIdentifier(
+                "tfodMonitorViewId", "id", hardwareMap.appContext.getPackageName());
+        TFObjectDetector.Parameters tfodParameters = new TFObjectDetector.Parameters(tfodMonitorViewId);
+        tfod = ClassFactory.getInstance().createTFObjectDetector(tfodParameters, vuforia);
+        tfod.loadModelFromAsset(TFOD_MODEL_ASSET, LABEL_GOLD_MINERAL, LABEL_SILVER_MINERAL);
+    }
```

Comment:
The proper way to do this would be to enhance the MineralDetectionTask so that it can use either the phone camera or a web camera.

Part of your problem here is that you tried to pull everything out of the MineralDetectionTask and that seems to have gone wrong.